packages/pulumi-sdk/src/ApplicationBuilder.ts

```diff
@@ -20,9 +21,10 @@ export interface ApplicationBuilderConfig extends Partial<ApplicationHooks> {
 }
 
 export interface ApplicationStack {
-    refresh(): Promise<void>;
-    preview(): Promise<void>;
-    up(): Promise<void>;
+    app?: PulumiApp;
+    refresh(): Promise<any>;
+    preview(): Promise<any>;
```

Comment:
Should this be `Promise<void>`?

Reply:
`stack.up()` in pulumi automation returns an object with a lot of data, that may be useful in a deploy script, so I basically passed the result back to the caller. Because Automation API behaves differently than our pulumi wrapper, I couldn't add any meaningful typings right now.
We don't use the result now, but I had some experiments with storage app state files migration and eventually left it like that. We can get rid of it though.

Reply:
We went through the code a bit.

We think it's better to make it generic / be able to specify the return type upon using the `up` and probably other methods (refresh/preview) as well.

So something like:

```ts
await up<{cognitoUserPoolId: string, dynamoDbTableNane: string, ....}>();
```

Does that make sense?

In any case, we both agreed that we should not use `any` nor `void`.

Reply:
Added types from pulumi automation API. For legacy version it will return undefined.